catalyst-gateway/bin/src/service/api/mod.rs

```diff
@@ -72,8 +79,34 @@ pub(crate) fn mk_api(
 
     // Add all the hosts where this API should be reachable.
     for host in hosts {
-        service = service.server(ServerObject::new(host));
+        service = service.server(ServerObject::new(host).description("API Host"));
     }
 
+    // Get localhost name
+    if let Ok(hostname) = gethostname().into_string() {
+        let hostname_with_port = format!("{hostname}:{PORT}");
+        service = service
+            .server(ServerObject::new(hostname_with_port).description("Server at localhost name"));
+    }
+
+    // Get local IP address v4 and v6
+    let network_interfaces = list_afinet_netifas();
+    if let Ok(network_interfaces) = network_interfaces {
+        for (name, ip) in &network_interfaces {
+            if *name == "en0" {
+                let (ip_with_port, desc) = match ip {
+                    IpAddr::V4(_) => {
+                        let ip_str = format!("{ip}:{PORT}");
+                        (ip_str, "Server at local IPv4 address")
+                    },
+                    IpAddr::V6(_) => {
+                        let ip_str = format!("[{ip}]:{PORT}");
+                        (ip_str, "Server at local IPv6 address")
+                    },
+                };
+                service = service.server(ServerObject::new(ip_with_port).description(desc));
+            }
+        }
+    }
```

Comment:
Nothing wrong with the code but the nesting is so deep. You could use if guard, or make match arms into a single line.

Reply:
Will change match arms into single line